### DISCUSSION TITLE
Does a comprehensive guide to ILSpy exist?

### DISCUSSION THREAD

**Discussion by MaxCookie** [icsharpcode/ILSpy#3304] [Help] [answered]
I was looking for a desktop app that will give me the same results as https://sharplab.io/, which is using ILSpy as far as i know. But i don't really understand how i should tune the options to see state machines of async or a anonymous delegate classes that compiler makes (basically elimination of syntax sugar plus actual code that is generated behind the scenes in C#), thanks in advance.

**Comment by greenozon** [comment]
I guess you are looking for this feature?

![image](https://github.com/user-attachments/assets/bd0b6887-d6ef-4631-ac6d-2e6d576fcc7a)

**Reply by MaxCookie** [reply]
> I guess you are looking for this feature?
Maybe, but i don have such a long list of options in my version (ILSpy version 9.0.0.7660-preview2), and in order to get what i want i need to select "no transforms"?

**Reply by siegfriedpammer** [reply]
"C# - no transforms" and friends are only available in debug builds. You don't want this because it will still contain transformed state machines and lambdas, which are transformed much earlier in the pipeline.

You can just select C# 1 as a version in the second dropdown and go to the decompilation options and uncheck all the boxes (of C# 1).

**Reply by MaxCookie** [reply]
@siegfriedpammer thanks, i did everything you've mentioned, and results are fine, except for the one thing - I can't see generated class that is mentioned in the Main method of my mini Console application (c__DisplayClass0_0 is obviously a generated class, but i can't see it anywhere) - is it even possible?

**Reply by siegfriedpammer** [reply, marked as answer]
Left of the language selection dropdown there is a group of buttons, which control the visibility of items in the tree view. By selecting the third option from the left you should see all the types and members. Then you should be able to select the compiler-generated display class in the tree view and/or navigate to it via link.

Note, that these types are usually nested types so you need to look into the type containing the method in question.

If it is still not visible that sounds like a bug, but it is hard to say without the actual binary to look at.

**Reply by MaxCookie** [reply]
@siegfriedpammer it worked, thank you!

**Comment by greenozon** [comment]
To get the latest CI build - go to the main page  https://github.com/icsharpcode/ILSpy
scroll down after files list you'll see Download section
click on [latest CI build (master)](https://github.com/icsharpcode/ILSpy/actions?query=workflow%3A%22Build+ILSpy%22+branch%3Amaster+is%3Asuccess+event%3Apush)
Click on latest (top) worklow line with green checkbox
scroll down
and select any version you like (I'm using debug build)

**Reply by siegfriedpammer** [reply]
It's generally not advised to use a debug build except for in depth testing like @greenozon does. Due to all the assertions being checked the decompilation speed is quite bad.

**Reply by greenozon** [reply]
yeah, agree, ;)
debug builds are special guys for special needs :)
